### DISCUSSION TITLE
Unable to introspect token generated by service users

### DISCUSSION THREAD

**Discussion by yerrasrinivas** [zitadel/zitadel#8658] [Q&A] [answered]
1). Created an service user

2). Token generated
![service_user](https://github.com/user-attachments/assets/634cf980-8383-4ca5-a2d9-af58d05c5d42)

2). Created API project with credentials.

3). Introspecting token always giving {active: false}

![introspect](https://github.com/user-attachments/assets/d2cfacc8-06b9-4a83-87ec-012328f38740)

**Comment by fforootd** [comment]
Hm what client do you use to authenticate for the introspect endpoint?

What happens if you send the same token to the userinfor endpoint?

**Reply by yerrasrinivas** [reply]
Hi,
I have created an API Application (OIDC) and Basic Authentication. Use it for Token introspect.

I tried UserInfo endpoint with the same API application and working fine.

![image](https://github.com/user-attachments/assets/19e2220c-1005-4feb-892f-660c2bced837)

**Reply by fforootd** [reply]
Not sure whats broken here let me ask around.

**Reply by livio-a** [reply]
Hey @yerrasrinivas 

Looking at the logs, it assume the issue is resolved, is that correct?
At least I could see that some days ago the following error: "token is not valid for this client" -> probably wrong projectID?
Since yesterday there were no such errors on the introspection endpoint and the returned body size was bigger.

**Reply by yerrasrinivas** [reply]
Hey @Livio,
Problem: Token Introspection failing for service users access token

Nope. Problem is not resolved, with ACCESS_TOKEN with service users.

But if i setup PAT Token for service users, and token introspect is giving response

**Reply by livio-a** [reply, marked as answer]
OK. I quickly checked your project setup. You need to request the projectID in the scope. You currently have the appID in there.

**Comment by yerrasrinivas** [comment]
Hi Livio,
Good catch.. Yes it working when I use project_id in scope..

Great help ...
thank you
